File: VRSampleApp/app/src/main/java/com/ooyala/sample/utils/AdList.java
package com.ooyala.sample.utils;


import android.content.Context;

import com.ooyala.sample.parser.VideoJSONParser;

import java.util.Arrays;
import java.util.List;

public class AdList {

  private static AdList instance;

  private List<VideoData> videoList = null;

  public static AdList getInstance() {
    if (instance == null) {
      instance = new AdList();
    }
    return instance;
  }

  public List<VideoData> getVideoList(Context context) {
    if (videoList == null) {
      videoList = new VideoJSONParser().getVideoData(context.getAssets());
    }
    return videoList;
  }

}

File: VRSampleAppKotlin/app/src/main/java/com/ooyala/sample/utils/AdList.kt
package com.ooyala.sample.utils

import android.content.Context
import com.ooyala.sample.parser.VideoJSONParser

class AdList {

  private var adList: List<VideoData>? = null

  companion object {
    val instance : AdList = AdList()
  }

  fun getVideoList(context: Context): List<VideoData> {
    if (adList == null) {
      adList = VideoJSONParser().getVideoList(context.assets)
    }
    return adList!!
  }
}
